src/charm.py

```diff
@@ -537,22 +542,62 @@ def _configure_pki_secrets_engine(self) -> None:  # noqa: C901
             private_key=str(private_key),
             mount=PKI_MOUNT,
         )
+        issued_certificates_validity = self._calculate_pki_certificates_validity_period(
+            provider_certificate.certificate
+        )
         if not vault.is_common_name_allowed_in_pki_role(
             role=PKI_ROLE_NAME,
             mount=PKI_MOUNT,
             common_name=config_common_name,
+        ) or issued_certificates_validity != vault.get_role_max_ttl(
+            role=PKI_ROLE_NAME, mount=PKI_MOUNT
         ):
             vault.create_or_update_pki_charm_role(
                 allowed_domains=config_common_name,
                 mount=PKI_MOUNT,
                 role=PKI_ROLE_NAME,
+                max_ttl=f"{issued_certificates_validity}s",
             )
         # Can run only after the first issuer has been actually created.
         try:
             vault.make_latest_pki_issuer_default(mount=PKI_MOUNT)
         except VaultClientError as e:
             logger.error("Failed to make latest issuer default: %s", e)
 
+    def _intermediate_ca_can_sign_certificates(
```

Comment:
I'd recommend going with exactly what this method does: `intermediate_ca_exceeds_role_ttl`

Reply:
✅